tests: "Using mocker in a with context is not supported"

With pytest-mock 2.0.0 I get this error for the tests:

```
[   54s] =================================== FAILURES ===================================
[   54s] _______________________________ test_from_graph ________________________________
[   54s] 
[   54s] mocker = <pytest_mock.plugin.MockFixture object at 0x7efc6d9fcc70>
[   54s] graph = <wiring.graph.Graph object at 0x7efc6d9fcb20>
[   54s] mock_plugin = <Mock spec='PluginManager' id='139622636045600'>
[   54s] mock_view = <Mock id='139622636047184'>
[   54s] 
[   54s]     def test_from_graph(mocker, graph, mock_plugin, mock_view):
[   54s]         mocker.patch("tomate.app.dbus.SessionBus")
[   54s]     
[   54s]         from tomate.app import Application
[   54s]     
[   54s]         graph.register_instance("tomate.view", mock_view)
[   54s]         graph.register_instance("tomate.plugin", mock_plugin)
[   54s]         graph.register_factory("tomate.app", Application)
[   54s]     
[   54s]         app = Application.from_graph(graph)
[   54s]     
[   54s]         assert isinstance(app, Application)
[   54s]     
[   54s] >       with mocker.patch(
[   54s]             "tomate.app.dbus.SessionBus.return_value.request_name",
[   54s]             return_value=dbus.bus.REQUEST_NAME_REPLY_EXISTS,
[   54s]         ):
[   54s] 
[   54s] tests/test_app.py:27: 
[   54s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   54s] /usr/lib/python3.8/site-packages/pytest_mock/plugin.py:197: in __call__
[   54s]     return self._start_patch(self.mock_module.patch, *args, **kwargs)
[   54s] /usr/lib/python3.8/site-packages/pytest_mock/plugin.py:159: in _start_patch
[   54s]     self._enforce_no_with_context(inspect.stack())
[   54s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   54s] 
[   54s] self = <pytest_mock.plugin.MockFixture._Patcher object at 0x7efc6d9fc250>
[   54s] stack = [FrameInfo(frame=<frame at 0x7efc6d9ac950, file '/usr/lib/python3.8/site-packages/pytest_mock/plugin.py', line 159, co...mbda>', code_context=['        self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(\n'], index=0), ...]
[   54s] 
[   54s]     def _enforce_no_with_context(self, stack):
[   54s]         """raises a ValueError if mocker is used in a with context"""
[   54s]         caller = stack[2]
[   54s]         frame = caller[0]
[   54s]         info = inspect.getframeinfo(frame)
[   54s]         if info.code_context is None:
[   54s]             # no source code available (#169)
[   54s]             return
[   54s]         code_context = " ".join(info.code_context).strip()
[   54s]     
[   54s]         if code_context.startswith("with mocker."):
[   54s] >           raise ValueError(
[   54s]                 "Using mocker in a with context is not supported. "
[   54s]                 "https://github.com/pytest-dev/pytest-mock#note-about-usage-as-context-manager"
[   54s]             )
[   54s] E           ValueError: Using mocker in a with context is not supported. https://github.com/pytest-dev/pytest-mock#note-about-usage-as-context-manager
[   54s] 
[   54s] /usr/lib/python3.8/site-packages/pytest_mock/plugin.py:178: ValueError
```

Looks like this is not supported (anymore): https://github.com/pytest-dev/pytest-mock#note-about-usage-as-context-manager

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -1,3 +1,4 @@
+import contextlib
 import dbus
 import pytest
 
@@ -24,10 +25,11 @@ def test_from_graph(mocker, graph, mock_plugin, mock_view):
 
     assert isinstance(app, Application)
 
-    with mocker.patch(
-        "tomate.app.dbus.SessionBus.return_value.request_name",
-        return_value=dbus.bus.REQUEST_NAME_REPLY_EXISTS,
-    ):
+    with contextlib.ExitStack() as stack:
+        stack.enter_context(mocker.patch(
+            "tomate.app.dbus.SessionBus.return_value.request_name",
+            return_value=dbus.bus.REQUEST_NAME_REPLY_EXISTS,
+        ))
         dbus_app = Application.from_graph(graph)
 
         assert isinstance(dbus_app, dbus.Interface)

